### ISSUE TITLE
Use null values for certain events table fields

### ISSUE THREAD

**Issue by darozak** [Darwin-River/Ex-Machinis#70]
The development website now displays a log of in-game events at http://advolition.com/in-game-event.  However, certain columns in this table (and the SQL DB) should be blank (i.e. null) depending on the type of event that is being reported.  This is because the data in certain columns is meaningless with certain types of event effects,  

For example, incrementing the inventory does not result in the spacecraft moving to the sun as a new location.  Nor will it reduce the player's credits to zero.  These fields should be blank (null in the DB) rather than zero.

Similarly, moving to a new location does not actually set quantity and credits to zero.  Rather these fields have no bearing on that type of event and need to be rendered as null in the database. 

Is it possible to modify the SQL DB or game engine so that all events table fields are initialized as null unless a value is deliberately added by either the game or event engine?  

Basically, a null has a different meaning than a zero in the events table.

**Comment by crodnun** [comment]
Changes for this are in place, ready for testing them. I have added modifications at both:

- Current DB content, to set to NULL those values that previously had a 0
- Game engine to deal with this requirement, keep NULL values when event does not affect the fiedl.

So far these are the fields affected by this new logic:

new_quantity | new_credits | new_location | new_transmission 

Just a minor comment, I noticed that dates here are incorrect:

![image](https://user-images.githubusercontent.com/16229970/73050933-aea56680-3e81-11ea-8085-e5074e5bad9d.png)

Hope you already have this in mind, or it has an explanation.

Feel free to stress this change out to confirm that my changes work as expected.
